src/js/tasks/dateDisplayFormatter.js

```diff
@@ -0,0 +1,231 @@
+Date.prototype.fromNow = function () {
+  let date = this;
+  let now = new Date();
+
+  let dYears = now.getFullYear() - date.getFullYear();
+  let dMonths = now.getMonth() - date.getMonth();
+  let dDays = now.getDate() - date.getDate();
+
+  if (dDays < 0) {
+    dMonths--;
+    dDays += new Date(date.getFullYear(), date.getMonth(), 0).getDate();
+  }
+
+  if (dMonths < 0) {
+    dYears--;
+    dMonths += 12;
+  }
+
+  let message = "";
+  if (dYears > 0) {
+    message += `${dYears} years `;
+  }
+
+  if (dMonths > 0) {
+    message += `${dMonths} months `;
+  }
+
+  if (dDays > 0) {
+    message += `${dDays} days `;
+  }
+
+  if (message.length > 0) {
+    message += "ago";
+  } else {
+    message = "Today";
+  }
+
+  return message;
+};
+
+const dateDisplayFormatter = (() => {
+  const monthNamesLong = [
+    "January",
+    "February",
+    "March",
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+  ];
+
+  const monthNamesNarrow = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+  ];
+
+  function getMonthPattern(format) {
+    if (format.indexOf("MMMM") != -1) {
+      return "MMMM";
+    } else if (format.indexOf("MMM") != -1) {
+      return "MMM";
+    } else if (format.indexOf("MM") != -1) {
+      return "MM";
+    } else {
+      return null;
+    }
+  }
+
+  function getDayPattern(format) {
+    if (format.indexOf("DD") != -1) {
+      return "DD";
+    } else {
+      return null;
+    }
+  }
+
+  function getYearPattern(format) {
+    if (format.indexOf("YYYY") != -1) {
+      return "YYYY";
+    } else if (format.indexOf("YY") != -1) {
+      return "YY";
+    } else {
+      return null;
+    }
+  }
+
+  function parseDay(input, inputFormat) {
+    let day;
+    let pattern = getDayPattern(inputFormat);
+    if (pattern == "DD") {
+      let dayIndex = inputFormat.indexOf("DD");
+      day = +input.slice(dayIndex, dayIndex + 2);
+    } else {
+      return null;
+    }
+
+    return day;
+  }
+
+  function parseMonth(input, inputFormat) {
+    let month;
+    let pattern = getMonthPattern(inputFormat);
+
+    switch (pattern) {
+      case "MMMM":
```

Comment:
you use those string more thana once so it is a signal to move them to constants.